canopeum_backend/canopeum_backend/models.py

```diff
@@ -150,32 +189,59 @@ def upload_to(_, filename):
 class Asset(models.Model):
     asset = models.FileField(upload_to=upload_to, null=False)
 
+    def delete(self, using=None, keep_parents=False):
+        self.asset.delete()
+        return super().delete()
+
 
 class Site(models.Model):
     name = models.TextField()
     is_public = models.BooleanField(blank=False, null=False, default=False)
     site_type = models.ForeignKey("Sitetype", models.DO_NOTHING, blank=True, null=True)
-    coordinate = models.ForeignKey(Coordinate, models.DO_NOTHING, blank=True, null=True)
+    coordinate = models.ForeignKey(Coordinate, models.SET_NULL, blank=True, null=True)
     description = models.TextField(blank=True, null=True)
     size = models.TextField(blank=True, null=True)
     research_partnership = models.BooleanField(blank=True, null=True)
     visible_map = models.BooleanField(blank=True, null=True)
     visitor_count = models.IntegerField(blank=True, null=True)
-    contact = models.ForeignKey(Contact, models.DO_NOTHING, blank=True, null=True)
-    announcement = models.ForeignKey(Announcement, models.DO_NOTHING, blank=True, null=True)
-    image = models.ForeignKey(Asset, models.DO_NOTHING, blank=True, null=True)
+    contact = models.ForeignKey(Contact, models.SET_NULL, blank=True, null=True)
+    announcement = models.ForeignKey(Announcement, models.SET_NULL, blank=True, null=True)
+    image = models.ForeignKey(Asset, models.SET_NULL, blank=True, null=True)
+
+    def delete(self, using=None, keep_parents=False):
```

Comment:
Is `keep_parents` there because it's an overload? If so:
`from typing import overload`
```suggestion
    @overload
    def delete(self, using=None, keep_parents=False):
```
That'll resolve `FBT002`